csharp/src/Ice/SignaledSocketStream.cs

```diff
@@ -88,46 +127,42 @@ protected ValueTask<T> WaitSignalAsync(CancellationToken cancel = default)
             {
                 Debug.Assert(_tokenRegistration == default);
                 cancel.ThrowIfCancellationRequested();
-                _tokenRegistration = cancel.Register(() =>
-                {
-                    try
-                    {
-                        _tokenRegistration.Token.ThrowIfCancellationRequested();
-                    }
-                    catch (Exception ex)
-                    {
-                        Abort(ex);
-                    }
-                });
+                _tokenRegistration = cancel.Register(() => Abort(new OperationCanceledException(cancel)));
             }
             return new ValueTask<T>(this, _source.Version);
         }
 
         T IValueTaskSource<T>.GetResult(short token)
         {
-            Debug.Assert(token == _source.Version);
+            bool lockTaken = false;
```

Comment:
I think this method should be listed with the public method and (unless I am missing something) written as
`public T GetResult(short token)`

Reply:
I didn't make it public because I didn't really see the use case for calling it directly. Why do you think it should be public rather than only callable through the interface?